mypy/test/data.py

```diff
@@ -270,11 +270,35 @@ def setup(self) -> None:
         self.tmpdir = tempfile.TemporaryDirectory(prefix='mypy-test-')
         os.chdir(self.tmpdir.name)
         os.mkdir(test_temp_dir)
+
+        # Precalculate steps for find_steps()
+        steps: Dict[int, List[FileOperation]] = {}
+
         for path, content in self.files:
-            dir = os.path.dirname(path)
-            os.makedirs(dir, exist_ok=True)
-            with open(path, 'w', encoding='utf8') as f:
-                f.write(content)
+            m = re.match(r'.*\.([0-9]+)$', path)
```

Comment:
I moved this stuff so I could combine the `re.match` logic.
I wonder if we can totally eliminate `find_steps` and just inject the steps as an arg to `suite.run_test()`